localgov_guides.module

```diff
@@ -52,6 +52,17 @@ function localgov_guides_node_insert(NodeInterface $node) {
  */
 function localgov_guides_node_update(NodeInterface $node) {
   if ($node->bundle() == 'localgov_guides_page') {
+    if (
+      ($previous = $node->original) &&
+      ($previous->localgov_guides_parent->target_id != $node->localgov_guides_parent->target_id)
+    ) {
+      if ($old_parent = $node->original->localgov_guides_parent->entity) {
+        foreach (array_keys($old_parent->localgov_guides_pages->getValue(), ['target_id' => $node->id()]) as $offset) {
+          $old_parent->localgov_guides_pages->offsetUnset($offset);
+        }
+        $old_parent->save();
+      }
+    }
```

Comment:
I am trying to understand what's going on here.  I have written down what I have understood so far.  Have I lost the plot?
```
$me = $node;
$my_parent = $me->localgov_guides_parent->target_id;
$me_in_a_previous_life = $me->original;
$my_previous_parent = $me_in_a_previous_life->localgov_guides_parent->target_id;
$has_my_parent_changed = ($my_parent !== $my_previous_parent);
$nah = !$has_my_parent_changed;
if ($nah) {
  goto end_of_story; // @see https://www.php.net/goto :D
}

$me_and_my_old_siblings = $my_previous_parent->localgov_guides_pages->getValue();
$old_me = array_keys($me_and_my_old_siblings, ['target_id' => $me->id()]);
leave_old_family($my_previous_parent, $old_me);

function leave_old_family($my_previous_parent, $old_me) {
  $my_previous_parent->localgov_guides_pages->offsetUnset($old_me);
  $my_previous_parent->save();
};

end_of_story:
```

Reply:
Hopefully https://github.com/localgovdrupal/localgov_guides/pull/27/commits/6731f937ae22c4ff25cdc655e28240a18117c507 helps, and will in the future too, it's not exactly obvious it is true.

Also this is what @stephen-cox had done for the same on Campaigns, I'd missed this before I wrote this one for guides https://github.com/localgovdrupal/localgov_campaigns/blob/ed71f9955d8c97efa17d10bf4bd3f65f232a6a03/localgov_campaigns.module#L54